### ISSUE TITLE
Roadmap for new release?

### ISSUE THREAD

**Issue by klodoma** [GeniusesOfSymfony/WebSocketBundle#386]
Hi, 

I just want to ask if there is a "roadmap" or an "idea" when a new release will be coming. I am using the dev-master version for the last months even for production, but our CI failed the last days due to last changes. 

I had to use the dev-master for symfony 4.2.*, I had several issues. 

My current work-around is to fork and use my repo which is a dev snapshot from february or so... 

All the best.

**Comment by mbabker** [comment]
I don't have a roadmap in mind personally.  The last couple of days, I made a lot of the classes final and added typehints to most methods, which is probably where your CI issues are coming from (and unfortunately depending on how in depth your integration is that'll take a little time to fix up in your app, a lot of my time yesterday was in just that for one of my clients).

In my mind, the last "big" things I want to do before I'd personally consider tagging a 2.0 beta is set up the CI for this repo to properly test the `ext/amqp` and `ext/zmq` pusher integrations, the rest of the bundle is now pretty well tested (compared to 1.x where there aren't many tests) and look into fixing the `ClientManipulatorInterface` so that its `findByUsername` method better accounts for multiple active connections for one user.  With the Symfony 4.3 release, there are some forward compat issues to look at regarding Symfony 5.0 (i.e. the event dispatcher changes, and how practical it'll be to do something that is compatible with 3.4 thru 5.0 or if 2.x will only support 3.4 LTS + 4.x then a 3.0 branch for 4.4 LTS + 5.x) and decisions to make on that.  Otherwise, I think the big things from here are just going to be cleaning up the documentation, backporting any needed changes to 1.x, and tagging a new 1.x release with all the needed deprecations and backports for 2.0.  Since I've been running `dev-master` in production for a client, I'm fairly comfortable with the bundle not being massively broken, just some of the upgrade steps are a little painful depending on what features you're using.

**Comment by klodoma** [comment]
Yes, I think the code-base is quite good for the master, I've been using it for the last few months. 
And yes, my CI issues are coming from the latest changes. 

I know that the clean-up takes time, that is always like that. 

In order not to use the dev-master till these happen, maybe we could create a branch or a "in-between-release" with this version: https://github.com/GeniusesOfSymfony/WebSocketBundle/commit/e2f61a50aa59733eedcfa7dbda87295c616f673d 

This will allow to use it in production, without forking and you can do the rest when you consider you have time. You can call it 2.0-pre-alpha or you name-it... :) it doesn't really matter...

**Comment by mbabker** [comment]
You can also lock to that commit in `composer.json`:

```json
{
    "require": {
        "gos/web-socket-bundle": "dev-master#e2f61a50aa59733eedcfa7dbda87295c616f673d"
    }
}
```

Should be the same thing as if I went back and branched or tagged from that commit, and should let you ignore any other changes in master until you've got the time to deal with the last batch of changes.

**Comment by klodoma** [comment]
> You can also lock to that commit in `composer.json`:
> 
> ```json
> {
>     "require": {
>         "gos/web-socket-bundle": "dev-master#e2f61a50aa59733eedcfa7dbda87295c616f673d"
>     }
> }
> ```
> 
> Should be the same thing as if I went back and branched or tagged from that commit, and should let you ignore any other changes in master until you've got the time to deal with the last batch of changes.

Even better, didn't know about this feature :) Thanks a lot!

and .... keep up the good work!!!
